### ISSUE TITLE
[image_picker] preferredCameraDevice: CameraDevice.front does not work  in getVideo

### ISSUE THREAD

**Issue by lukemorse** [flutter/flutter#74674]
It seems to work for getImage, but not getVideo.  I have tested this on my iPhone 11.  

## Steps to Reproduce

1. Add image_picker to pubspec.yaml file: 
`dependencies:
  image_picker: ^0.6.7+21`

2. Add NSPhotoLibraryUsageDescription, NSCameraUsageDescription, and NSMicrophoneUsageDescription to info.plist
3. Call this code: `PickedFile _file = await ImagePicker().getVideo(
        source: ImageSource.camera,
        preferredCameraDevice: CameraDevice.front
        );`

**Expected results:** Camera opens using front-facing camera

**Actual results:** Camera opens using rear-facing camera

<details>
  <summary>Logs</summary>

```
```

```
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.7 19H114 darwin-x64, locale en)
    • Flutter version 1.22.5 at /Users/luke/Development/flutter
    • Framework revision 7891006299 (7 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/luke/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/luke/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/luke/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.0

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin installed
    • Dart plugin version 201.9317
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[!] Connected device
    ! No devices available

! Doctor found issues in 1 category.
```

```
```

</details>

**Comment by TahaTesser** [comment]
Hi @lukemorse 
I can reproduce but when I initialize the image picker then calling it `preferredCameraDevice` works fine, I noticed official example does the same way 

Try this code sample blow.

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:image_picker/image_picker.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  final picker = ImagePicker();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: Container(
          child: Text('Hello World'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () async {
          final pickedFile = await picker.getImage(
              source: ImageSource.camera,
              preferredCameraDevice: CameraDevice.front);
        },
      ),
    );
  }
}
```
</details>
Hope this helps.
Closing, as this isn't an issue with `image_picker` itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by lukemorse** [comment]
Hi @TahaTesser, thank you for your response.  Sorry, I should have posted a more comprehensive code sample.  Your code does work, but again my issue is with getVideo and _not_ getImage.  If you try to run your code again replacing getImage with getVideo, I think you'll find that it does not use the preferredCameraDevice.

**Comment by TahaTesser** [comment]
Hi @lukemorse 
Thanks for that, issue reproduces with `getVideo`, while testing the issue, I switch between,  thought it was the same behavior.


<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:image_picker/image_picker.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  final picker = ImagePicker();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: Container(
          child: Text('Hello World'),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        child: Icon(Icons.add),
        onPressed: () async {
          final pickedFile = await picker.getVideo(
              source: ImageSource.camera,
              preferredCameraDevice: CameraDevice.front);
        },
      ),
    );
  }
}
```
</details>




<details>
<summary>logs</summary>

```bash
[        ]   Platform: remote-ios
[        ]  Connected: no
[        ]   SDK Path: "/Users/tahatesser/Library/Developer/Xcode/iOS DeviceSupport/14.3 (18C66) arm64e/Symbols"
[        ] (lldb)     target create
"/Users/tahatesser/AndroidStudioProjects/triage-examples/flutter_plugin_examples/image_picker_example/build/ios/iphoneos/Runner.app"
[+6721 ms] Current executable set to
'/Users/tahatesser/AndroidStudioProjects/triage-examples/flutter_plugin_examples/image_picker_example/build/ios/iphoneos/Runner.app'
(arm64).
[        ] (lldb)     script
fruitstrap_device_app="/private/var/containers/Bundle/Application/80C88D36-D8ED-4618-964D-88559ADAF54E/Runner.app"
[ +590 ms] (lldb)     script fruitstrap_connect_url="connect://127.0.0.1:63155"
[        ] (lldb)     script fruitstrap_output_path=""
[        ] (lldb)     script fruitstrap_error_path=""
[        ] (lldb)     target modules search-paths add /usr "/Users/tahatesser/Library/Developer/Xcode/iOS DeviceSupport/14.3 (18C66)
arm64e/Symbols/usr" /System "/Users/tahatesser/Library/Developer/Xcode/iOS DeviceSupport/14.3 (18C66) arm64e/Symbols/System"
"/private/var/containers/Bundle/Application/80C88D36-D8ED-4618-964D-88559ADAF54E"
"/Users/tahatesser/AndroidStudioProjects/triage-examples/flutter_plugin_examples/image_picker_example/build/ios/iphoneos"
"/var/containers/Bundle/Application/80C88D36-D8ED-4618-964D-88559ADAF54E"
"/Users/tahatesser/AndroidStudioProjects/triage-examples/flutter_plugin_examples/image_picker_example/build/ios/iphoneos" /Developer
"/Users/tahatesser/Library/Developer/Xcode/iOS DeviceSupport/14.3 (18C66) arm64e/Symbols/Developer"
[  +24 ms] (lldb)     command script import "/tmp/178E7C1A-0CE9-4145-8A7B-732E537DA666/fruitstrap_00008020_001059882212002E.py"
[   +5 ms] (lldb)     command script add -f fruitstrap_00008020_001059882212002E.connect_command connect
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_00008020_001059882212002E.run_command run
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_00008020_001059882212002E.autoexit_command autoexit
[        ] (lldb)     command script add -s asynchronous -f fruitstrap_00008020_001059882212002E.safequit_command safequit
[        ] (lldb)     connect
[  +57 ms] (lldb)     run
[ +343 ms] success
[        ] Application launched on the device. Waiting for observatory port.
[+1669 ms] Observatory URL on device: http://127.0.0.1:65373/
[   +8 ms] Attempting to forward device port 65373 to host port 63163
[        ] executing: /Users/tahatesser/Code/flutter_beta/bin/cache/artifacts/usbmuxd/iproxy 63163:65373 --udid
00008020-001059882212002E
[+1024 ms] Forwarded port ForwardedPort HOST:63163 to DEVICE:65373
[   +1 ms] Forwarded host port 63163 to device port 65373 for Observatory
[   +4 ms] Installing and launching... (completed in 32.1s)
[   +3 ms] Caching compiled dill
[ +108 ms] Connecting to service protocol: http://127.0.0.1:63163/
[   +2 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at
http://127.0.0.1:63163/.
[ +148 ms] DDS is listening at http://127.0.0.1:63165/8o34_vQj1zc=/.
[  +77 ms] Successfully connected to service protocol: http://127.0.0.1:63163/
[  +17 ms] DevFS: Creating new filesystem on the device (null)
[  +14 ms] DevFS: Created new filesystem on the device
(file:///private/var/mobile/Containers/Data/Application/50B9E7DA-4D75-412B-A2A9-E3D7B4405972/tmp/image_picker_exampleuQJv4R/image_pi
cker_example/)
[   +3 ms] Updating assets
[ +132 ms] Syncing files to device Taha’s iPhone...
[   +1 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +2 ms] <- recompile package:triage/main.dart 615efed5-45f9-4014-b4d0-2cc2e28b5f85
[        ] <- 615efed5-45f9-4014-b4d0-2cc2e28b5f85
[ +166 ms] Updating files.
[        ] DevFS: Sync finished
[        ] Syncing files to device Taha’s iPhone... (completed in 172ms)
[        ] Synced 0.0MB.
[   +1 ms] <- accept
[  +15 ms] Connected to _flutterView/0x106832420.
[   +1 ms] Flutter run key commands.
[   +3 ms] r Hot reload. 🔥🔥🔥
[   +1 ms] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] An Observatory debugger and profiler on Taha’s iPhone is available at: http://127.0.0.1:63165/8o34_vQj1zc=/
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[+3528 ms] [VERBOSE-2:profiler_metrics_ios.mm(203)] Error retrieving thread information: (os/kern) object terminated
[+3277 ms] [Camera] Failed to read exposureBiasesByMode dictionary: Error Domain=NSCocoaErrorDomain Code=4864 "***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL" UserInfo={NSDebugDescription=***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL}
[+2542 ms] [Camera] Failed to read exposureBiasesByMode dictionary: Error Domain=NSCocoaErrorDomain Code=4864 "***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL" UserInfo={NSDebugDescription=***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL}
[+1690 ms] [Camera] Failed to read exposureBiasesByMode dictionary: Error Domain=NSCocoaErrorDomain Code=4864 "***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL" UserInfo={NSDebugDescription=***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL}
[+4265 ms] [Camera] Failed to read exposureBiasesByMode dictionary: Error Domain=NSCocoaErrorDomain Code=4864 "***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL" UserInfo={NSDebugDescription=***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL}
[+1858 ms] [Camera] Failed to read exposureBiasesByMode dictionary: Error Domain=NSCocoaErrorDomain Code=4864 "***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL" UserInfo={NSDebugDescription=***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL}
[+3831 ms] [Camera] Failed to read exposureBiasesByMode dictionary: Error Domain=NSCocoaErrorDomain Code=4864 "***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL" UserInfo={NSDebugDescription=***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL}
[+2570 ms] [Camera] Failed to read exposureBiasesByMode dictionary: Error Domain=NSCocoaErrorDomain Code=4864 "***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL" UserInfo={NSDebugDescription=***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL}
[+2476 ms] [Camera] Failed to read exposureBiasesByMode dictionary: Error Domain=NSCocoaErrorDomain Code=4864 "***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL" UserInfo={NSDebugDescription=***
-[NSKeyedUnarchiver _initForReadingFromData:error:throwLegacyExceptions:]: data is NULL}
[+1588 ms] [hcln] AVHapticClient.mm:416:-[AVHapticClient finish:]: Player was not running - bailing with error Error
Domain=com.apple.CoreHaptics Code=-4805 "(null)" for client 0x10033a6
[   +8 ms] [Feedback] core haptics engine finished for <_UIFeedbackCoreHapticsIgnoreCaptureHapticsOnlyEngine: 0x2806157a0> with
error: Error Domain=com.apple.CoreHaptics Code=-4805 "(null)"
```
</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 1.22.6, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.22.6 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 9b2d32b605 (5 days ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (2 available)
    • RMX2001 (mobile)     • EUYTFEUSQSRGDA6D          • android-arm64 • Android 10 (API 29)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios           • iOS 14.3

! Doctor found issues in 1 category.
```
```bash
[✓] Flutter (Channel master, 1.26.0-13.0.pre.194, on macOS 11.1 20C69 darwin-x64, locale en-GB)
    • Flutter version 1.26.0-13.0.pre.194 at /Users/tahatesser/Code/flutter_master
    • Framework revision 71aec53acb (5 hours ago), 2021-01-26 21:53:04 -0800
    • Engine revision b11bef83a5
    • Dart version 2.12.0 (build 2.12.0-263.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • RMX2001 (mobile)     • EUYTFEUSQSRGDA6D          • android-arm64  • Android 10 (API 29)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios            • iOS 14.3
    • macOS (desktop)      • macos                     • darwin-x64     • macOS 11.1 20C69 darwin-x64
    • Chrome (web)         • chrome                    • web-javascript • Google Chrome 88.0.4324.96

• No issues found!
```

</details>

Thank you
